test/tests/feedsTest.js

```diff
@@ -1,9 +1,46 @@
-describe("Zotero.Feeds", function () {
+describe.only("Zotero.Feeds", function () {
 	
 	after(function* () {
 		yield clearFeeds();
 	});
 	
+	describe('#importFromOPML', function() {
+		var opmlUrl = getTestDataUrl("feeds.opml");
+		var opmlString;
+		
+		before(function* (){
+			opmlString = yield Zotero.File.getContentsFromURLAsync(opmlUrl)
+		});
+		
+		beforeEach(function* () {
+			yield clearFeeds();
+		});
+		
+		it('imports feeds correctly', function* (){
+			let shouldExist = {
+				"http://example.com/feed1.rss": "A title 1",
+				"http://example.com/feed2.rss": "A title 2",
+				"http://example.com/feed3.rss": "A title 3",
+				"http://example.com/feed4.rss": "A title 4"
+			};	yield Zotero.Feeds.importFromOPML(opmlString);
+			let feeds = Zotero.Feeds.getAll();
+			for (let feed of feeds) {
+				assert.equal(shouldExist[feed.url], feed.name, "Feed exists and title matches");
+				delete shouldExist[feed.url];
+			}
+			assert.equal(Object.keys(shouldExist).length, 0, "All feeds from opml have been created");
+		});
+		
+		it("doesn't fail if some feeds already exist", function* (){
+			yield createFeed({url: "http://example.com/feed1.rss"});
+			try{
+				yield Zotero.Feeds.importFromOPML(opmlString)
+			} catch(e) {
+				assert.fail(e)
+			};
```

Comment:
Don't need the try/catch here — as long as there's a yield it will fail properly without it.